docs/profiling.md

```diff
@@ -0,0 +1,92 @@
+# vHive profiling tool
+
+## Methodology
+
+The tool is for benchmarking vHive instances by recording hardware counters, requests per second
+ (RPS) and execution time for functions running inside instances.
+
+The tool includes following components: 
+- A loader function loads requests to vHive round by round (increment 5% of input RPS each round 
+  by default) until input RPS and profiles performance. if tail latency violates 10x image 
+  unloaded service time at a step, the function returns the metric before it.
+- A single latency measurement goroutine mimics a rare event loader that injects a request least 
+  every 500ms and measures runtimes of requests to compute mean latency and 90 percentile latency.
+- A bind socket function binds all VMs to socket 1 (only compatible with a 2x 16-core machine or 
+  above).
+- A profiler invokes [toplev](https://github.com/andikleen/pmu-tools) to collect hardware counters.
+- A plotter plots line charts for a list of metrics.
+
+A load step divides into three parts: warm-up, profiling and cool-down. During the profile 
+period, the benchmark records the average execution time of invocations and how many invocations 
+return successfully. During the profile period, it also invokes [pmu-tools](https://github.com/andikleen/pmu-tools) 
+to profile user-defined counters followed by [TopDown method](https://ieeexplore.ieee.org/document/6844459). 
+After the measurement finishes, completed RPS per core, average execution time and the average 
+counters will save in the `profile.csv`.
+
+## Runtime Arguments
+```
```

Comment:
please add the default values, I think Golang prints them if you run a test with `-h`.

Reply:
Done